futures/tests/abortable.rs

```diff
@@ -0,0 +1,39 @@
+extern crate futures;
+
+use futures::channel::oneshot;
+use futures::executor::block_on;
+use futures::future::{abortable, Aborted};
+
+use std::thread;
+use std::time::Duration;
+
+#[test]
+fn abortable_works() {
+    let (_tx, a_rx) = oneshot::channel::<()>();
+    let (abortable_rx, abort_handle) = abortable(a_rx);
+
+    abort_handle.abort();
+    assert_eq!(Err(Aborted), block_on(abortable_rx));
+}
+
+#[test]
+fn abortable_awakens() {
+    let (_tx, a_rx) = oneshot::channel::<()>();
+    let (abortable_rx, abort_handle) = abortable(a_rx);
+
+    thread::spawn(move || {
+        thread::sleep(Duration::from_millis(100));
```

Comment:
Tests should not use `sleep`. Here is some (untested) code that should do the same:

```rust
let (tx1, rx1) = mpsc::channel::<()>();
let (tx2, rx2) = mpsc::channel::<()>();

let (fut, abort_handle) = future::poll_fn(move |cx| {
    tx1.send(()).unwrap();
    Poll::Pending
}).abortable()

fut.map_err(err => tx2.send(err).unwrap())
    .run_in_background();

assert_eq!((), rx1.recv().unwrap()); // Future inside `Abortable` runs

assert_eq!(Err(mpsc::TryRecvError::Empty), rx2.try_recv()); // Not aborted yet
abort_handle.abort(); // `abort` awakens the `Abortable` future
assert_eq!(Aborted, rx2.recv().unwrap()) // `Abortable` future has completed
```

Reply:
Replaced this with a test using an AtomicBool counter as a waker.